src/frontend/src/handler/alter_parallelism.rs

```diff
@@ -76,6 +76,13 @@ pub async fn handle_alter_parallelism(
                 session.check_privilege_for_drop_alter(schema_name, &**table)?;
                 table.id.table_id()
             }
+            StatementType::ALTER_SOURCE => {
+                let (source, schema_name) =
+                    reader.get_source_by_name(db_name, schema_path, &real_table_name)?;
+
+                session.check_privilege_for_drop_alter(schema_name, &**source)?;
```

Comment:
Do we need to add a check for whether the source is a shared one.

Reply:
you are right. Thanjs for catching this